iot-bootc-image.sh

```diff
@@ -0,0 +1,274 @@
+        ;;
+    "fedora-43")
+        OS_VARIANT="fedora-unknown"
+        OCI_ARCHIVE="Fedora-IoT-bootc-${ARCH}-43.${COMPOSE_ID}.ociarchive"
+        ;;
+    "fedora-44")
+        OS_VARIANT="fedora-rawhide"
+        OCI_ARCHIVE="Fedora-IoT-bootc-${ARCH}-44.${COMPOSE_ID}.ociarchive"
+        ;;
+    *)
+        log_error "Unsupported distro: ${ID}-${VERSION_ID}"
+        exit 1
+        ;;
+esac
+
+# Download OS image
+download_image() {
+    log_info "Downloading OS image..."
+
+    local image_url="${COMPOSE_URL}/${OCI_ARCHIVE}"
+    if [[ -f "${OCI_ARCHIVE}" ]]; then
+        log_info "Container image already exists, skipping download"
+        return 0
+    fi
+    
+    if ! sudo wget --progress=bar:force "${image_url}"; then
+        log_error "Failed to download image from ${image_url}"
+        exit 1
+    fi
+
+    if [[ -f "${OCI_ARCHIVE}" ]]; then
+        log_success "Download completed: ${OCI_ARCHIVE}"
+    else
+        log_error "Downloaded file not found: ${OCI_ARCHIVE}"
+        exit 1
+    fi
+}
+
+# Wait for SSH to be available
+wait_for_ssh() {
+    local ip_address=$1
+    local max_attempts=30
+    local attempt=0
+    
+    log_info "Waiting for SSH on ${ip_address}..."
+    
+    while [[ ${attempt} -lt ${max_attempts} ]]; do
+        if ssh "${SSH_OPTIONS[@]}" -i "${SSH_KEY}" "${EDGE_USER}@${ip_address}" 'echo -n "READY"' 2>/dev/null | grep -q "READY"; then
+            log_success "SSH is ready"
+            return 0
+        fi
+        
+        attempt=$((attempt + 1))
+        sleep 10
+    done
+    
+    log_error "SSH connection timed out after $((max_attempts * 10)) seconds"
+    return 1
+}
+
+# Main execution
+log_info "Starting fedora IoT bootc image test"
+log_info "Test UUID: ${TEST_UUID}"
+log_info "Temporary directory: ${TEMPDIR}"
+
+# Execute steps
+download_image
+
+# Container image settings
+OCI_ARCHIVE_TAG="${VERSION_ID}"
+
+log_info "Copying container image into storage with a controlled tag"
+sudo skopeo copy oci-archive:"${OCI_ARCHIVE}" containers-storage:"${CONTAINER_IMG_NAME}:${OCI_ARCHIVE_TAG}"
+
+log_info "Preparing Containerfile"
+tee Containerfile > /dev/null << STOPHERE
+FROM ${CONTAINER_IMG_NAME}:${OCI_ARCHIVE_TAG}
+RUN echo 'root' | passwd --stdin root
+# RUN useradd -m -G wheel admin \
+#  && install -d -m 700 -o admin -g admin /home/admin/.ssh \
+#  && printf '%s\n' $SSH_KEY_PUB > /home/admin/.ssh/authorized_keys \
+#  && chown admin:admin /home/admin/.ssh/authorized_keys \
+#  && chmod 600 /home/admin/.ssh/authorized_keys \
+#  && sed -i 's/^#\?PasswordAuthentication .*/PasswordAuthentication no/' /etc/ssh/sshd_config \
+#  && sed -i 's/^#\?PermitRootLogin .*/PermitRootLogin prohibit-password/' /etc/ssh/sshd_config
+
+STOPHERE
+
+log_info "Build and push container image"
+sudo podman build -f Containerfile -t "${QUAY_REPO_TAG}"
+sudo podman push "${CONTAINER_IMG_NAME}:${OCI_ARCHIVE_TAG}" --creds="${QUAY_USERNAME}:${QUAY_PASSWORD}" "${QUAY_REPO_TAG}"
```

Comment:
I would suggest we can skip the quay part. 
Just download bootc container from koji and convert it into qcow2 image.